### DISCUSSION TITLE
toktx: Specify pixel format

### DISCUSSION THREAD

**Discussion by brianpopow** [KhronosGroup/KTX-Software#378] [Q&A] [answered]
Is it possible to specify the pixel format when using the `toktx` tool? I could not find a option for that. Maybe i am missing something.

I am currently working on writing a decoder for ktx2 textures and it would be very useful to generate a test set of all the different pixel formats, to see if it actually works with all of them.

**Comment by MarkCallow** [comment, marked as answer]
No it is not. `toktx` determines the texture format from the pixel format of the input data. I do not want to add general purpose format conversions so am not planning to add such an option so at present you are limited to the formats that can be stored in JPEG, PNG or Netpbm files and the compressed formats to which those may be encoded (ETC1S/BasisLZ and UASTC). I may add other block-compression encoders in the future which would enlarge the number of formats in which you can create textures.
